src/cls/IPM/Utils/Module.cls

```diff
@@ -1233,12 +1238,20 @@ ClassMethod LoadNewModule(
 
         set pDirectory = ##class(%File).NormalizeDirectory("",pDirectory)
 
+        set newModuleObj = ..GetModuleObjectFromPath(pDirectory,.foundNewModuleObj)
+        // Note that for update/install commands, the operation version validity has already been checked in %IPM.Main.Install().
+        if isLoad && foundNewModuleObj {
+            // Check that the module currently installed is not being downgraded. Also check that "load" is not being used to update a module, unless the -force flag is passed.
+            do ..CheckOperationVersionValidity(newModuleObj, .pParams)
```

Comment:
I just realized one thing: we may want to confirm at some point in this flow when providing a path, that the path provided, matches the module name provided for update. A user could otherwise accidentally have provided an incorrect tarball and update the wrong module unintentionally

Reply:
Yes checking exactly that. You will need to pass that info down to here from Update since that context is not provided for downstream use when -path is provided right now.